R/geom-raster.r

```diff
@@ -52,8 +52,20 @@ GeomRaster <- ggproto("GeomRaster", Geom,
     hjust <- params$hjust %||% 0.5
     vjust <- params$vjust %||% 0.5
 
-    w <- resolution(data$x, FALSE)
-    h <- resolution(data$y, FALSE)
+    x_diff <- diff(sort(unique(data$x)))
+    if (!length(unique(x_diff)) == 1) {
```

Comment:
I think this assumption is too strong. For example:

``` r
unique(diff(c(1:10 / 10)))
#> [1] 0.1 0.1 0.1 0.1
```

<sup>Created on 2019-04-24 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

We need a version of `unique()` with tolerance.